modules/charon-core/src/test/java/org/wso2/charon3/core/protocol/endpoints/MeResourceManagerTest.java

```diff
@@ -0,0 +1,1553 @@
+
+    }
+
+    private SCIMResponse getEncodeSCIMExceptionObject(AbstractCharonException exception) {
+
+        JSONEncoder encoder = new JSONEncoder();
+        Map<String, String> responseHeaders = new HashMap<>();
+        responseHeaders.put(SCIMConstants.CONTENT_TYPE_HEADER, SCIMConstants.APPLICATION_JSON);
+        return new SCIMResponse(exception.getStatus(), encoder.encodeSCIMException(exception), responseHeaders);
+    }
+
+    private User getNewUser() throws BadRequestException, CharonException, InternalErrorException {
+
+        SCIMResourceTypeSchema schema = SCIMResourceSchemaManager.getInstance().getUserResourceSchema();
+        JSONDecoder decoder = new JSONDecoder();
+
+        return decoder.decodeResource(newUserSCIMObjectString, schema, new User());
+    }
+
+    @DataProvider(name = "dataForGetSuccess")
+    public Object[][] dataToGetSuccess() throws CharonException, InternalErrorException, BadRequestException {
+
+        User user = getNewUser();
+        String id = user.getId();
+        String name = user.getUserName();
+
+        return new Object[][]{
+                {id, name, null, null, success, user},
+                {id, name, null, "emails", success, user},
+                {id, name, "username,meta", null, success, user},
+                {id, name, "username", "emails", success, user}
+        };
+    }
+
+    @Test(dataProvider = "dataForGetSuccess")
+    public void testGetSuccess(String id, String name, String attributes,
+                               String excludeAttributes, int expectedScimResponseStatus, Object objectUser)
+            throws BadRequestException, NotFoundException, CharonException {
+
+        User user = (User) objectUser;
+
+        SCIMResourceTypeSchema schema = SCIMResourceSchemaManager.getInstance().getUserResourceSchema();
+
+        Map<String, Boolean> requiredAttributes = ResourceManagerUtil.getOnlyRequiredAttributesURIs(
+                (SCIMResourceTypeSchema) CopyUtil.deepCopy(schema), attributes, excludeAttributes);
+        mockStatic(AbstractResourceManager.class);
+
+        when(AbstractResourceManager.getResourceEndpointURL(SCIMConstants.USER_ENDPOINT))
+                .thenReturn(endpoint);
+        when(AbstractResourceManager.getEncoder()).thenReturn(new JSONEncoder());
+        when(AbstractResourceManager.getDecoder()).thenReturn(new JSONDecoder());
+        when(userManager.getMe(name, requiredAttributes)).thenReturn(user);
+
+        SCIMResponse outputScimResponse = meResourceManager.get(name, userManager, attributes, excludeAttributes);
+        JSONObject obj = new JSONObject(outputScimResponse.getResponseMessage());
+
+        //Assertions
+
+        Assert.assertEquals(outputScimResponse.getResponseStatus(), expectedScimResponseStatus);
+        Assert.assertEquals(obj.getString("userName"), name);
+        Assert.assertEquals(obj.getString("id"), id);
+
+        String returnedURI = outputScimResponse.getHeaderParamMap().get("Location");
```

Comment:
Use SCIMConstants.LOCATION_HEADER for "Location". check all applicable places

Reply:
Fixed in commit d8f6f1c18502d1174c97a568b559f353a6b945a6